pkg/object/unstructured.go

```diff
@@ -18,49 +18,26 @@ var (
 	crdGK       = schema.GroupKind{Group: "apiextensions.k8s.io", Kind: "CustomResourceDefinition"}
 )
 
-// UnstructuredsToObjMetas converts a slice of unstructureds to a slice of
-// ObjMetadata. If the values for any of the unstructured objects doesn't
-// pass validation, an error will be returned.
-func UnstructuredsToObjMetas(objs []*unstructured.Unstructured) ([]ObjMetadata, error) {
-	objMetas := make([]ObjMetadata, 0, len(objs))
-	for _, obj := range objs {
-		objMeta, err := UnstructuredToObjMeta(obj)
-		if err != nil {
-			return nil, err
-		}
-		objMetas = append(objMetas, objMeta)
-	}
-	return objMetas, nil
-}
-
-// UnstructuredsToObjMetasOrDie converts a slice of unstructureds to a slice of
-// ObjMetadata. If the values for any of the unstructured objects doesn't
-// pass validation, the function will panic.
-func UnstructuredsToObjMetasOrDie(objs []*unstructured.Unstructured) []ObjMetadata {
-	objMetas, err := UnstructuredsToObjMetas(objs)
-	if err != nil {
-		panic(err)
+// UnstructuredToObjMetadata converts Unstructured to ObjMetadata.
+func UnstructuredToObjMetadata(obj *unstructured.Unstructured) ObjMetadata {
+	return ObjMetadata{
+		GroupKind: obj.GroupVersionKind().GroupKind(),
+		Name:      obj.GetName(),
+		Namespace: obj.GetNamespace(),
 	}
-	return objMetas
-}
-
-// UnstructuredToObjMeta extracts the identifying information from an
-// Unstructured object and returns it as Objmetadata. If the values doesn't
-// pass validation, an error will be returned.
-func UnstructuredToObjMeta(obj *unstructured.Unstructured) (ObjMetadata, error) {
-	return CreateObjMetadata(obj.GetNamespace(), obj.GetName(),
-		obj.GroupVersionKind().GroupKind())
 }
 
 // UnstructuredToObjMetaOrDie extracts the identifying information from an
 // Unstructured object and returns it as Objmetadata. If the values doesn't
 // pass validation, the function will panic.
+// TODO: Deprecate in favor of handling the error to impl continue on error.
 func UnstructuredToObjMetaOrDie(obj *unstructured.Unstructured) ObjMetadata {
```

Comment:
Even with the changes in this PR, there are still 127 usages of UnstructuredToObjMetaOrDie in this repo. 

A lot of them are in tests and can probably be replaced by UnstructuredToObjMetadata, but the rest are in the code and could cause a panic if executed on a code path during a SkipInvalid apply/destroy.

Reply:
I'm not sure if it's a good idea to try to replace them all in this PR or not.

Reply:
This refactor was moved to https://github.com/kubernetes-sigs/cli-utils/pull/505